src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -702,102 +748,332 @@ int fx_muxer_t::execute(
 
 namespace
 {
-    int get_delegate_from_runtime(
-        const pal::string_t& impl_dll_dir,
-        corehost_init_t* init,
-        coreclr_delegate_type type,
-        void** delegate)
+    int get_init_info_for_component(
+        const host_startup_info_t &host_info,
+        host_mode_t mode,
+        pal::string_t &runtime_config_path,
+        /*out*/ pal::string_t &hostpolicy_dir,
+        /*out*/ std::unique_ptr<corehost_init_t> &init)
     {
-        pal::dll_t corehost;
-        hostpolicy_contract host_contract{};
-        corehost_get_delegate_fn coreclr_delegate = nullptr;
+        // Read config
```

Comment:
After switching to not using the full init struct for secondary contexts, I made this stop creating the `corehost_init_t` since as you pointed out we don't care about most of that. (It is still named `get_init_info_for_secondary_component`, since I was thinking of init info as referring to the idea, not the actual class, but I could see that being ambiguous, so a clearer name would be welcome). I think the duplication here is less interesting now (and also just less in quantity).